myframe.cpp

```diff
@@ -607,6 +607,23 @@ int MyFrame::GetTextWidth(wxControl *pControl, const wxString& string)
     return width;
 }
 
+// Get either timelapse value or state-dependent variable-exposure-delay
+int MyFrame::GetExposureDelay()
+{
+    if (!m_varDelayConfig.enabled)
+        return m_timeLapse;
+
+    int rslt = 0;
+    if (pGuider->IsGuiding() && PhdController::IsIdle() && !pGuider->IsRecentering() && pMount->GetGuidingEnabled())   // pMount will be valid in any state of guiding
+        rslt = m_varDelayConfig.longDelay;
+    else
+        rslt = m_varDelayConfig.shortDelay;
+    if (rslt != m_varLastExpDelay)
+        Debug.Write(wxString::Format("Exposure delay set to %d\n", rslt));
+    m_varLastExpDelay = rslt;

```

Comment:
`m_varLastExpDelay` is uninitialized, so it would be better if we explicitly initialize it somewhere.
Also, there is no logging when m_varDelayConfig.enabled changes from true to false.
A possible solution to both of these problems would be to change m_varLastExpDelay from a member variable to a static variable used only in this function. Something like this:
```
// Get either timelapse value or state-dependent variable-exposure-delay, in milliseconds
int MyFrame::GetExposureDelay()
{
    int rslt;

    if (!m_varDelayConfig.enabled)
        rslt = m_timeLapse;
    else if (pGuider->IsGuiding() && PhdController::IsIdle() && !pGuider->IsRecentering() &&
             pMount->GetGuidingEnabled())
    {
        rslt = m_varDelayConfig.longDelay;
    }
    else
        rslt = m_varDelayConfig.shortDelay;

    static int s_lastExposureDelay = -1;
    if (rslt != s_lastExposureDelay)
    {
        Debug.Write(wxString::Format("Exposure delay set to %d\n", rslt));
        s_lastExposureDelay = rslt;
    }

    return rslt;
}
```